vpr/src/place/net_cost_handler.cpp

```diff
@@ -88,14 +94,33 @@ static vtr::vector<ClusterNetId, NetUpdateState> bb_updated_before; // [0...clus
  * layer_ts_bb_coord_new are used.
  */
 
-/* [0...cluster_ctx.clb_nlist.nets().size()-1] -> 3D bounding box*/
-static vtr::vector<ClusterNetId, t_bb> ts_bb_coord_new, ts_bb_edge_new;
-/* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> 2D bonding box on a layer*/
-static vtr::vector<ClusterNetId, std::vector<t_2D_bb>> layer_ts_bb_edge_new, layer_ts_bb_coord_new;
-/* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> number of sink pins on a layer*/
-static vtr::Matrix<int> ts_layer_sink_pin_count;
-/* [0...num_afftected_nets] -> net_id of the affected nets */
-static std::vector<ClusterNetId> ts_nets_to_update;
+struct TSInfo{
+    /* [0...cluster_ctx.clb_nlist.nets().size()-1] -> 3D bounding box*/
+    vtr::vector<ClusterNetId, t_bb> ts_bb_coord_new, ts_bb_edge_new;
+    /* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> 2D bonding box on a layer*/
+    vtr::vector<ClusterNetId, std::vector<t_2D_bb>> layer_ts_bb_edge_new, layer_ts_bb_coord_new;
+    /* [0...cluster_ctx.clb_nlist.nets().size()-1][0...num_layers-1] -> number of sink pins on a layer*/
+    vtr::Matrix<int> ts_layer_sink_pin_count;
+    /* [0...num_afftected_nets] -> net_id of the affected nets */
+    std::vector<ClusterNetId> ts_nets_to_update;
+};
+static struct TSInfo ts_info;
+
+/**
+ * @brief This class is used to hide control flows needed to distinguish 2d and 3d placement
+ */
+class BB2D3DControlFlow {
+    bool cube_bb = false; 
```

Comment:
Ew global state. I know why this is here, it prevents passing parameters around. @amin1377 this is your call. Do you ever foresee cube_bb changing during execution of the VPR flow? Or is it truly a static variable that never changes.

Reply:
The bounding box type should be determined at the beginning of the flow, and I don't expect it to change during execution.

Reply:
I see. Just wanted to check.